drivers/i2c/i2c_mcux_flexcomm.c

```diff
@@ -193,23 +203,50 @@ static int mcux_flexcomm_transfer(const struct device *dev,
 }
 
 #if defined(CONFIG_I2C_TARGET)
+
+static int mcux_flexcomm_target_find_free_target(struct mcux_flexcomm_data *data)
+{
+	for (int i = 0; i < MCUX_FLEXCOMM_MAX_SLAVES; i++) {
+		if (!data->targets[i].target_attached) {
+			return i;
+		}
+	}
+	return -EBUSY;
+}
+
+static int mcux_flexcomm_target_find_target_by_addr(struct mcux_flexcomm_data *data,
```

Comment:
find_target_... sounds like this should maybe return a target (or a NULL pointer) instead of the index?

Reply:
Isn't that backwards? The index is not really interesting, how about making _find_free_target() and and _find_target_by_addr simply return a pointer to struct mcux_flexcomm_target_config or NULL?

Reply:
reworked to `_find_free_target`